guides/plugins/apps/tax-provider.md

```diff
@@ -0,0 +1,140 @@
+# Tax provider
+
+Tax calculations differ from country to country. Especially in the US, the sales tax calculation can be tedious, as the laws and regulation differ from state to state or even county or certain city taxes can apply. Therefore, most shops use a third-party service (so called tax-providers) to calculate sales taxes for them.
+
+Starting with version 6.5.0.0, Shopware allows apps to integrate custom tax calculation, which could include an automatic tax calculation with a tax provider. An app has to provide an endpoint, which is called during checkout and provide new tax rates. The requests and responses of all of your endpoints will be signed and feature JSON content.
+
+## Prerequisites
+
+You should be familiar with the concept of Apps and their registration.
+
+{% page-ref page="app-base-guide.md" %}
+
+To reproduce this example, you should also be aware of how to set up an app on your development platform.
+
+{% page-ref page="local-development/" %}
+
+## Manifest configuration
+
+To signalize Shopware, that your app uses a custom tax calculation, you have to provide one or more `tax-provider` properties inside a `tax` parent property in your app's `manifest.xml`.
+
+Below, you can see an example definition of a working tax-provider.
+
+{% code title="manifest.xml" %}
+
+```markup
+<?xml version="1.0" encoding="UTF-8"?>
+<manifest xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://raw.githubusercontent.com/shopware/platform/trunk/src/Core/Framework/App/Manifest/Schema/manifest-1.0.xsd">
+    <meta>
+        <!-- The name of the app should not change. Otherwise all payment methods are created as duplicates. -->
+        <name>PaymentApp</name>
+        <!-- ... -->
+    </meta>
+    
+    <tax>
+        <tax-provider>
+            <identifier>myCustomTaxProvider</identifier>                        <!-- Unique identifier of the tax provider -->
+            <name>My custom tax provider</name>                                 <!-- Display name of the tax provider -->    
+            <priority>1</priority>                                              <!-- Priority of the tax provider - can be changed in the administration as well -->
+            <processUrl>https://tax-provider.app/provide-taxes</processUrl>     <!-- Url of your implementation - is called during checkout to provide taxes -->
+        </tax-provider>
+    </tax>
+</manifest>
+```
+
+{% endcode %}
+
+After successful installation of your app, the tax provider will already be used during checkout to provide taxes.
+
+## Tax provider endpoint
+
+During checkout, Shopware checks for any active tax providers - sorted by priority - and will call the `processUrl` to provide taxes one-by-one, until one of endpoint successfully provides taxes for the current cart.
```

Comment:
This sounds like a lot to do for Shopware during the checkout. To not break the checkout I assume there is a HTTP client timeout, that this endpoint has to fulfil. What is the response time I need to make sure I can provide? In the admin it is like 5 seconds, what is it here?

Reply:
We use the same global timeout for every app system request via guzzle.
Connection timeout is defined as 1s and general timeout after 5s. 

This can take its time at the checkout, especially if you are using multiple tax providers as fallback and get multiple connection timeouts, etc, but we have no other choice as to simply wait 🤷 . Furthermore, the tax provision is stopped immediately, as soon as the first tax provider is returning some tax data.

Reply:
Then I'd like to see like a warning block saying: if you are slower than 5s in your response the response than your tax provider will have no effect anymore. Or similar. Just to set some expectations.

Reply:
Then I'd like to see like a warning block saying: if you are slower than 5s in your response the response than your tax provider will have no effect anymore. Or similar. Just to set some expectations.